src/Plutarch/Extra/Precompile.hs

```diff
@@ -196,13 +197,17 @@ infixl 7 ###
 
 infixl 7 ###~
 
-
 {- | Convert 'CompiledTerm' into Plutarch 'Term'
- This can then be used to `plift` directly.
-
+ This can then be used to `plift`.
 -}
-toPTerm :: forall p. CompiledTerm p -> ClosedTerm p
-toPTerm prog = Term $ const $ pure $ TermResult (RCompiled $ UPLC._progTerm $ unScript . debugScript . toDScript $ prog) []
+toPTerm ::
+    forall (p :: S -> Type).
+    CompiledTerm p ->
+    ClosedTerm p
+toPTerm prog =
+    let (Script (UPLC.Program _ version term)) = debugScript . toDScript $ prog
```

Comment:
Why does this use `debugScript`? Pretty sure it should use `finalEvalDScript`, like `pliftCompiled'`.

Reply:
Hmm and unless we can get the official `plift` to output trace messages, maybe it should only use `script`. I'll ask if we could get that change in.